x/ibc/04-channel/keeper/packet.go

```diff
@@ -377,6 +380,8 @@ func (k Keeper) AcknowledgePacket(
 		k.SetNextSequenceAck(ctx, packet.GetDestPort(), packet.GetDestChannel(), nextSequenceAck+1)
 	}
 
+	k.deletePacketCommitment(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
```

Comment:
Wait, was this not already happening? This is in the spec - https://github.com/cosmos/ics/tree/master/spec/ics-004-channel-and-packet-semantics#acknowledgements

Reply:
(beyond avoiding cleanup, this needs to happen to prevent acknowledgement replay)

Reply:
found a similar spec discrepancy #6301. I'll update that one in a separate pr